code/game/gamemodes/antag_spawner.dm

```diff
@@ -53,15 +53,18 @@
 	if(loc == H || (in_range(src, H) && istype(loc, /turf)))
 		H.set_machine(src)
 		if(href_list["school"])
-			if (used)
+			if(used)
 				H << "You already used this contract!"
 				return
-			var/list/candidates = pollCandidates("Do you want to play as a wizard's [href_list["school"]] apprentice?", ROLE_WIZARD, null, ROLE_WIZARD, 150)
+			var/list/candidates = pollCandidatesForMob("Do you want to play as a wizard's [href_list["school"]] apprentice?", ROLE_WIZARD, null, ROLE_WIZARD, 150, src)

```

Comment:
this is a sleep so you probably need to check qdeleted and src after, otherwise you'll spawn antags into nullspace